controllers/metal3machine_controller.go

```diff
@@ -278,7 +331,7 @@ func (r *Metal3MachineReconciler) Metal3ClusterToMetal3Machines(o handler.MapObj
 		if m.Spec.InfrastructureRef.Name == "" {
 			continue
 		}
-		name := client.ObjectKey{Namespace: m.Namespace, Name: m.Name}
+		name := client.ObjectKey{Namespace: m.Spec.InfrastructureRef.Namespace, Name: m.Name}
```

Comment:
This also needs defaulting, and it should not be m.Name but m.Spec.InfrastructureRef.Name

Reply:
This is a bug as you reported, fixing it in a separate PR